boot/dune

```diff
@@ -0,0 +1,8 @@
+; This library must have no dependencies except config, as it is a
+; dependency for most Coq command line tools.
```

Comment:
Shouldn't it be fine to depend on `lib`?

Reply:
There would be no problem, however coqdep / coqdoc etc... depend on this, so it adds a bit of latency in the build.

I'd prefer for now to keep the deps lean, let's see how things look once we factor loadpath out.

Reply:
Another point in favor of having this more self-contained, is that eventually language servers and build systems would like to vendor this library as to avoid re-doing all the loadpath handling, etc...